content/docs/developing-operators/tasks.md

```diff
@@ -32,6 +32,12 @@ KUDO will apply all the listed resources and wait for all of them to become heal
 
 **Note:** current implementation will apply all resources in the given order so that e.g. a Pod can mount previously created ConfigMap. However, this is not part of the specification and might change in the future (all resources can be applied concurrently). If you need [happens-before-guarantee](https://en.wikipedia.org/wiki/Happened-before) between your resources, use e.g. multiple serial steps.
 
+The apply-task enhances deployed resources with KUDO specific labels and annotations. For resources that create or deploy pods (StatefulSets, Deployments, Daemonsets, etc. ) KUDO adds a special annotation to the pod template spec that is the hash from all resources used by the pod template spec (ConfigMaps and Secrets). The effect of this hash is that an update to an operator parameter that changes a ConfigMap triggers a restart of the pods that uses that ConfigMap.
+
+**Note:** This pod restart control will happen when the parent resource (StatefulSet, Deployment) is applied the next time. This can happen in the same task as the dependency, a task that is executed later in the same plan, or in a different plan executed at a later time. The pod restarts will not happen until the parent resource is applied again.
```

Comment:
After reading this paragraph a few times, I'm still not sure what is stated here.

Reply:
Rephrased that, hope it's a little bit clearer?

Reply:
Yes, got it now, thank you!